skhep/math/vectors.py

```diff
@@ -306,3 +308,236 @@ def __repr__(self):
+
+    @e.setter
+    def e(self, value):
+        """Sets e, aka momentum coordinate at position 3."""
+        self.__values[3] = value
+
+    @property
+    def m(self):
+        """Return the invariant mass."""
+        return self.mag
+
+    @property
+    def m2(self):
+        """Return the square of the invariant mass."""
+        return self.mag2
+
+    @property
+    def mass(self):
+        """Return the invariant mass."""
+        return self.m
+
+    @property
+    def mass2(self):
+        """Return the square of the invariant mass."""
+        return self.m2
+
+    def transversemass(self):
+        """Return the transverse mass."""
+        raise NotImplementedError
+
+    def beta(self):
+        """Return :math:`\\beta = v/c`."""
+        raise NotImplementedError
+
+    def gamma(self):
+        """Return :math:`\\gamma = 1/\\sqrt{1-\\beta^2}`."""
+        raise NotImplementedError
+
+    def eta(self):
+        """Return the pseudorapidity."""
+        raise NotImplementedError
+
+    def pseudorapidity(self):
+        """"Return the pseudorapidity. Alternative to eta() method."""
+        return self.eta()
+
+    def rapidity(self):
+        """Return the rapidity."""
+        raise NotImplementedError
+
+    @property
+    def mag(self):
+        """Magnitude, a.k.a. norm, of the Lorentz vector."""
+        mag2 = self.mag2
+        return ( sqrt(mag2) if mag2 >= 0. else -sqrt(-mag2) )
+
+    @property
+    def mag2(self):
+        """Square of the magnitude, a.k.a. norm, of the Lorentz vector."""
+        raise NotImplementedError
+
+    def set(self, x, y, z, t):
+        """Update/set all components at once."""
+        self.__values = [x, y, z, t]
+
+    def setpx(self, px):
+        """Update/set the px component."""
+        self.__values[0] = x
+
+    def setpy(self, py):
+        """Update/set the py component."""
+        self.__values[1] = y
+
+    def setpz(self, pz):
+        """Update/set the pz component."""
+        self.__values[2] = z
+
+    def sete(self, e):
+        """Update/set the energy/time component."""
+        self.__values[3] = e
+
+    def dot(self, other):
+        """Dot product with another Lorentz vector."""
+        raise NotImplementedError
+
+    def isspacelike(self):
+        """Check if Lorentz vector is space-like."""
+        return self.mag2 < 0.
+
+    def istimelike(self):
+        """Check if Lorentz vector is time-like."""
+        return self.mag2 > 0.
+
+    def islightlike(self):
+        """Check if Lorentz vector is light-like."""
+        return self.mag2 == 0.
+
+    def __repr__(self):
+        """Class representation."""
+        return "<LorentzVector (x={0},y={1},z={2},t={3})>".format(self.__values[0],
+                                                                  self.__values[1],
+                                                                  self.__values[2],
+                                                                  self.__values[3])
```

Comment:
I think a simple `*self.__values` would be enough. Otherwise, better write `self.x` etc. to make it more explicit.